06.bowling_object/frame.rb

```diff
@@ -3,23 +3,50 @@
 require_relative 'shot'
 
 class Frame
-  attr_reader :first_shot, :second_shot, :third_shot
+  attr_reader :shots, :frame_number
 
-  def initialize(first_shot, second_shot, third_shot = nil)
-    @first_shot = Shot.new(first_shot)
-    @second_shot = Shot.new(second_shot)
-    @third_shot = Shot.new(third_shot)
+  def initialize(frame_number, first_shot, second_shot, third_shot = nil)
+    @frame_number = frame_number
+    @shots = [
+      Shot.new(first_shot),
+      Shot.new(second_shot),
+      Shot.new(third_shot)
+    ]
   end
 
   def strike?
-    first_shot.score == 10
+    shots[0].score == 10
   end
 
   def spare?
-    !strike? && first_shot.score + second_shot.score == 10
+    !strike? && shots[0].score + shots[1].score == 10
```

Comment:
```suggestion
    !strike? && shots.first(2).sum(&:score) == 10
```

とすればシンプルかつDRYになりそうです